### PR TITLE
2d 1d binding

### PR DESCRIPTION
**Pull request by starcksophie**
• Added the  mr2d1d_trans function to the bindings
• Added the associated tests
• fixed wrong bord in transform.hpp

### CODE REVIEW

File: sparse2d/python/numpydata.hpp [hunk 1]
```diff
@@ -67,36 +67,50 @@ py::array_t<float> image2array_3d(const fltarray &image){
   for (int i=0; i<image.nx(); i++) {
     for (int j=0; j<image.ny(); j++) {
       for(int k=0; k<image.nz(); k++){
-        pointer[k + image.nz() * (j + i * image.ny())] = image(i, j, k);
+        pointer[i + image.nx() * (j + k * image.ny())] = image(i, j, k);
+        // pointer[k + image.nz() * (j + i * image.ny())] = image(i, j, k);
```
Comment (sfarrens) [inline sparse2d/python/numpydata.hpp]: Clean up commented lines.

File: sparse2d/python/pysparse.cpp [hunk 2]
```diff
@@ -15,6 +15,9 @@ for details.
 #include "transform_3D.hpp"
 #include "filter.hpp"
 #include "deconvolve.hpp"
+#include "mr_2d1d.hpp"
+// #include "mr2d1d.hpp"
```
Comment (sfarrens) [inline sparse2d/python/pysparse.cpp]: Clean up commented lines.

File: sparse2d/python/numpydata.hpp [hunk 3]
```diff
@@ -67,36 +67,50 @@ py::array_t<float> image2array_3d(const fltarray &image){
...
-  for (int i=0; i<array.shape(0); i++) {
-    for (int j=0; j<array.shape(1); j++) {
-      for (int k=0; k<array.shape(2); k++){
-        image(i, j, k) = pointer[k + array.shape(2) * (j + i * array.shape(1))];
+    for (int i=0; i<array.shape(2); i++) {
+      for (int j=0; j<array.shape(1); j++) {
+        for (int k=0; k<array.shape(0); k++){
+          image(i, j, k) = pointer[i + array.shape(2) * (j + k * array.shape(1))];
+          // image(i, j, k) = pointer[k + array.shape(2) * (j + i * array.shape(1))];
+        }
```
[the remark this answers is not in the document]
Comment (starcksophie) [reply sparse2d/python/numpydata.hpp]: Yes, it was used for the other bindings, the problem fixed here was only on the z coordinate.

File: pysap/extensions/sparse2d.py [hunk 4]
```diff
@@ -131,3 +133,45 @@ def show(self):  # pragma: no cover
         if self.data is None:
             raise AttributeError("The data must be deconvolved first !")
         self.data.show()
+
+
+class MR2D1D():
+    """ Define the structure that will be used to
+        store the deconvolution result.
```
[the remark this answers is not in the document]
Comment (starcksophie) [reply pysap/extensions/sparse2d.py]: No, this was an oversight.

**Pull request merged**